### ISSUE TITLE
SIGABRT RFC3339DateTool (Crash)

### ISSUE THREAD

**Issue by SudoPlz** [bugsnag/bugsnag-react-native#173]
Our app crash today because of bugsnag:

```javascript
SIGABRT
    /usr/lib/system/libsystem_kernel.dylib __pthread_kill
    /usr/lib/system/libsystem_pthread.dylib pthread_kill
    /usr/lib/system/libsystem_c.dylib abort
    /usr/lib/system/libsystem_malloc.dylib _szone_error
    /usr/lib/system/libsystem_malloc.dylib _free_list_checksum_botch
    /usr/lib/system/libsystem_malloc.dylib _tiny_malloc_from_free_list
    /usr/lib/system/libsystem_malloc.dylib _szone_malloc_should_clear
    /usr/lib/system/libsystem_malloc.dylib _malloc_zone_malloc
    /usr/lib/system/libsystem_malloc.dylib _malloc
    /usr/lib/libicucore.A.dylib icu::UVector::_init(int, UErrorCode&)
    /usr/lib/libicucore.A.dylib icu::UVector::UVector(void (*)(void*), signed char (*)(UElement, UElement), int, UErrorCode&)
    /usr/lib/libicucore.A.dylib icu::UnicodeSet::allocateStrings(UErrorCode&)
    /usr/lib/libicucore.A.dylib icu::UnicodeSet::UnicodeSet(icu::UnicodeString const&, UErrorCode&)
    /usr/lib/libicucore.A.dylib uprv_decNumberVersion
    /usr/lib/libicucore.A.dylib uprv_decNumberVersion
    /usr/lib/libicucore.A.dylib uprv_decNumberVersion
    /usr/lib/libicucore.A.dylib icu::DecimalFormat::construct(UErrorCode&, UParseError&, icu::UnicodeString const*, icu::DecimalFormatSymbols*)
    /usr/lib/libicucore.A.dylib icu::DecimalFormat::DecimalFormat(icu::UnicodeString const&, icu::DecimalFormatSymbols*, UNumberFormatStyle, UErrorCode&)
    /usr/lib/libicucore.A.dylib icu::NumberFormat::makeInstance(icu::Locale const&, UNumberFormatStyle, signed char, UErrorCode&)
    /usr/lib/libicucore.A.dylib icu::LocaleCacheKey<icu::SharedNumberFormat>::createObject(void const*, UErrorCode&) const
    /usr/lib/libicucore.A.dylib icu::UnifiedCache::_get(icu::CacheKeyBase const&, icu::SharedObject const*&, void const*, UErrorCode&) const
    /usr/lib/libicucore.A.dylib icu::NumberFormat::makeInstance(icu::Locale const&, UNumberFormatStyle, signed char, UErrorCode&)
    /usr/lib/libicucore.A.dylib icu::NumberFormat::makeInstance(icu::Locale const&, UNumberFormatStyle, signed char, UErrorCode&)
    /usr/lib/libicucore.A.dylib icu::NumberFormat::createSharedInstance(icu::Locale const&, UNumberFormatStyle, UErrorCode&)
    /usr/lib/libicucore.A.dylib icu::NumberFormat::createInstance(icu::Locale const&, UNumberFormatStyle, UErrorCode&)
    /usr/lib/libicucore.A.dylib icu::SimpleDateFormat::initialize(icu::Locale const&, UErrorCode&)
    /usr/lib/libicucore.A.dylib icu::SimpleDateFormat::SimpleDateFormat(icu::Locale const&, UErrorCode&)
    /usr/lib/libicucore.A.dylib icu::DateFormat::create(icu::DateFormat::EStyle, icu::DateFormat::EStyle, icu::Locale const&)
    /usr/lib/libicucore.A.dylib udat_open
    Frameworks/CoreFoundation.framework/CoreFoundation ___cficu_udat_open
    Frameworks/CoreFoundation.framework/CoreFoundation ___ResetUDateFormat
    Frameworks/CoreFoundation.framework/CoreFoundation _CFDateFormatterCreate
    Frameworks/Foundation.framework/Foundation -[NSDateFormatter _regenerateFormatter]
    Frameworks/Foundation.framework/Foundation -[NSDateFormatter stringForObjectValue:]
    /tmp/sandbox/workspace/node_modules/bugsnag-react-native/cocoa/vendor/kscrash/Source/KSCrash/Recording/Tools/RFC3339DateTool.m:49:12 +[RFC3339DateTool stringFromDate:]
    /tmp/sandbox/workspace/node_modules/bugsnag-react-native/cocoa/vendor/kscrash/Source/KSCrash/Recording/Tools/KSJSONCodecObjC.m:515:25 ksjsoncodecobjc_i_encodeObject
    /tmp/sandbox/workspace/node_modules/bugsnag-react-native/cocoa/vendor/kscrash/Source/KSCrash/Recording/Tools/KSJSONCodecObjC.m:497:26 ksjsoncodecobjc_i_encodeObject
    /tmp/sandbox/workspace/node_modules/bugsnag-react-native/cocoa/vendor/kscrash/Source/KSCrash/Recording/Tools/KSJSONCodecObjC.m:554:18 +[KSJSONCodec encode:options:error:]
    /tmp/sandbox/workspace/node_modules/bugsnag-react-native/cocoa/vendor/kscrash/Source/KSCrash/Recording/KSSystemInfo.m:457:47 kssysteminfo_toJSON
    /tmp/sandbox/workspace/node_modules/bugsnag-react-native/cocoa/vendor/kscrash/Source/KSCrash/Recording/KSCrashC.c:156:38 kscrash_install
    /tmp/sandbox/workspace/node_modules/bugsnag-react-native/cocoa/vendor/kscrash/Source/KSCrash/Recording/KSCrash.m:301:27 -[KSCrash install]
    /tmp/sandbox/workspace/node_modules/bugsnag-react-native/cocoa/vendor/bugsnag-cocoa/Source/BugsnagNotifier.m:164:10 -[BugsnagNotifier start]
    /tmp/sandbox/workspace/node_modules/bugsnag-react-native/cocoa/BugsnagReactNative.m:227:9 -[BugsnagReactNative startWithOptions:]
    Frameworks/CoreFoundation.framework/CoreFoundation ___invoking___
    Frameworks/CoreFoundation.framework/CoreFoundation -[NSInvocation invoke]
    Frameworks/CoreFoundation.framework/CoreFoundation -[NSInvocation invokeWithTarget:]
    /tmp/sandbox/workspace/node_modules/react-native/React/Base/RCTModuleMethod.m:521:3 -[RCTModuleMethod invokeWithBridge:module:arguments:]
    /tmp/sandbox/workspace/node_modules/react-native/React/CxxModule/RCTNativeModule.mm:95:17 invokeInner
    /tmp/sandbox/workspace/node_modules/react-native/React/CxxModule/RCTNativeModule.mm:67:5 operator()
    /usr/lib/system/libdispatch.dylib __dispatch_call_block_and_release
    /usr/lib/system/libdispatch.dylib __dispatch_client_callout
    /usr/lib/system/libdispatch.dylib __dispatch_queue_drain
    /usr/lib/system/libdispatch.dylib __dispatch_queue_invoke
    /usr/lib/system/libdispatch.dylib __dispatch_client_callout
    /usr/lib/system/libdispatch.dylib __dispatch_root_queue_drain
    /usr/lib/system/libdispatch.dylib __dispatch_worker_thread3
    /usr/lib/system/libsystem_pthread.dylib _pthread_wqthread
```

### Version
`"bugsnag-react-native": "2.3.2",`

Please fix this.

Thank you.

**Comment by fractalwrench** [comment]
Thanks for the report @SudoPlz - are you aware of any reproduction steps, or is this something that's happening sporadically? I'll investigate and see if I can find out what's causing the crash.

**Comment by SudoPlz** [comment]
We received this crash from your tool heheh, no idea what caused it but [here's the link to the issue](https://app.bugsnag.com/acuity-scheduling-inc/acuity-scheduling-mobile-app/errors/59f2a8ee0ae3060019b43beb?event_id=59f2a8ee0ae3060019b43bea&i=tl&m=ci) in bugsnag dashboard.

**Comment by fractalwrench** [comment]
Fixed in v2.5.3

**Comment by SudoPlz** [comment]
@fractalwrench You're the best, thank you very much.
